### PR TITLE
feat: penetration rate estimation, lot amenities, polling & expanded test coverage

### PR DESCRIPTION
**Pull request by chuckmilton**
## Summary

This PR introduces a **penetration rate estimation system** that converts raw device counts into real-occupancy estimates, adds a **TypeScript academic calendar module** for ML feature computation, adds a **lot amenities detail view** in the mobile app, implements **polling and AppState refresh** for real-time data, fixes **UI layout bugs**, and significantly **expands test coverage** across both backend and mobile.

---

### Three-Layer Estimation Service (penetration-estimation.service.ts)
1. **Academic calendar base rate**: derives expected daily commuters from the current semester/period (fall: 35K, spring: 34K, summer: 8K, session: 3K, break: 1.5K)
2. **Timezone-aware time multipliers**: adjusts commuters by time-of-day and day-of-week, with campus closure detection
3. **Activity-based scaling caps**: limits scaling factor based on campus-wide device activity to prevent over-inflation when few devices are active

Uses raw SQL `COUNT(DISTINCT device_hash)` for efficient device counting. Includes an empty-lot floor to prevent 0% estimates for lots that simply lack app users.

### Academic Calendar Module (academic-calendar.ts)
- TypeScript port of academic_calendar.py — pure rule-based heuristics, no database tables
- Computes semester dates dynamically using CSULB's scheduling patterns (fall = 4th Monday of August, spring = Tuesday after MLK Day, etc.)
- Exports `getSemester()`, `getWeekOfSemester()`, `isCampusOpen()`, `getExpectedCommuters()`, and more
- 86 unit tests covering all public APIs, multi-year consistency (2020–2027), and federal observed-holiday edge cases

### Snapshot ML Feature Columns
- `createSnapshots()` now populates `semester`, `academic_period`, `week_of_semester`, and `is_campus_open` at write time
- Values are lowercase strings matching the Python ML pipeline convention (e.g., `fall`, `regular`, `midterms`)
- ML training reads these pre-computed columns directly — no recomputation needed

### Database Changes
- Migration: `20260308045807_add_penetration_estimation` — adds `estimated_occupancy`, `penetration_rate_used`, `semester`, `academic_period`, `week_of_semester` to `occupancy_snapshots`
- Migration: `20260308200000_remove_calendar_tables` — drops `academic_calendar`, `campus_closures`, `campus_activity_baselines` tables (replaced by algorithmic computation)

### Service Integration
- `LotsService` returns `estimated_occupancy`, `estimated_available`, `raw_occupancy`, and `effective_penetration_rate` in `ParkingLotResponse`
- Lots service uses estimation-first filtering (e.g., `min_available` uses estimated values)
- `OccupancyEventsService` computes calendar features once per snapshot batch
- `PenetrationEstimationService` wired into both `LotsModule` and `OccupancyEventsModule`

### Seed Data
- Scaled all 28 lots' `current_occupancy` to raw device counts (`real_occupancy × penetration_rate`)
- Seeded 1,500 unique deterministic device hashes distributed proportionally across lots

---

### New `LotAmenities` Component
- Shows 5 info sections: lot overview, permits & rates, hours, special spaces (EV charging, accessible, motorcycle), and safety features
- Renders features as styled chips with conditional display logic
- Integrated below the hourly chart in `ShortTermForecastScreen`

---

## Mobile: Polling & Real-Time Data

- `useLotsList` / `useAllLotsData`: 30-second polling interval with `AppState` foreground refresh
- `useLotData`: 60-second polling interval with `AppState` foreground refresh
- Added `estimated_occupancy`, `estimated_available`, `raw_occupancy`, and `effective_penetration_rate` to `ParkingLotResponse` type
- Updated `MapScreen` and `RecommendationModal` tests for the new response shape

---

## Mobile: UI Bug Fixes

- **RecommendationModal**: Added `flex`/`flexShrink` to lot name and percentage badge so long names (e.g., "Pyramid Parking Structure") truncate instead of pushing the badge offscreen
- **ReliabilityModal**: Increased bottom padding to clear the iPhone home indicator

---

**Tests**: Backend 320 tests (19 suites) · Mobile 220 tests (21 suites)

### CODE REVIEW

**Review by lawrenceDegoma — APPROVED** [review]
Chuck,

Great work with the penetration rate estimation and its corresponding modal. Always exceptional work from you sir. If I could give you a higher score than a 5 on peer evals, I certainly would. As for a certain principal custodial engineer on the team however...

File: apps/backend/src/lots/academic-calendar.ts [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+  return {
+    semesterStart: start, semesterEnd: end,
+    classesStart: start, classesEnd: end,
+    finalsStart: null, finalsEnd: null,
+    readingDays: [],
+    breaks,
+  };
+}
+
+function generateSummer(may: SemesterInfo, year: number): SemesterInfo {
```
Comment (DottL) [inline apps/backend/src/lots/academic-calendar.ts]: There is a minor detail I want to point out regarding the pattern of summer. There is always an **overlap** between the last 2 week of may-intersession and summer-session.
- Summer starts in the last Tuesday of May. This overlaps with the last 2 weeks of may-intersession.

For ML, `academic_calendar.py` chose to prioritize may-intersession over summer-session in that overlap. This is not a huge issue since the model can learn patterns from the remaining features to fill in the gap. 

However, since you have a static lookup table to map the semester categories to a fixed commuter count, this would be a lot more punishing to the penetration rate estimation given the underestimation in that overlap. In this case, I think it'd be best if we pick summer-session over may-intersession for the overlap.
- I'd generate summer independently of may-intersession and swap the `SEMESTER_ORDER`
```  
function generateSummer(year: number): SemesterInfo {
  // 12-week session starting last Tuesday of May
  const start = lastWeekday(year, 5, 2); // Tuesday = 2 in UTC day convention
  const end = addDays(start, 83); // 12 weeks
```
```
const SEMESTER_ORDER: SemesterKey[] = ['fall', 'winter', 'spring', 'summer', 'mayIntersession'];
```

**Review by DottL — APPROVED** [review]
I left a comment for you to take a look at but everything else looks good boss 😸😸

Image

**Review by zachfpadilla — APPROVED** [review]
The penetration rate estimation system & new amenities modal both look great, as do the new tests & couple of UI fixes you implemented. And as always, I appreciate the fact that you took a second to update the docs.

Great job on the new academic calendar system too—it's a very smart implementation. I wonder how managed to identify those patterns? We could use something like that for the ML model, maybe @DottL should take a look at what you did.

Excellent work!

File: apps/backend/src/lots/academic-calendar.ts (hunk 1 above)
Comment (chuckmilton) [reply apps/backend/src/lots/academic-calendar.ts]: I appreciate you for pointing that out @DottL! I'll go make the changes right away 👍

**Comment by chuckmilton** [comment]
Thank you @lawrenceDegoma, @zachfpadilla, and @DottL for the feedback! I'm glad this change landed well with the team. The calendar heuristics are actually pretty clean once you reverse-engineer CSULB's academic schedule patterns. Good catch that it could be useful for our ML model!
